### ISSUE TITLE
Error parsing useEffects hooks dependency array 

### ISSUE THREAD

**Issue by leandrowilliam** [eslint/eslint#13421]
* **ESLint Version:**7.2.0
* **Node Version:**12.16.3
* **npm Version:**6.14.5

**What parser (default, Babel-ESLint, etc.) are you using?**
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "jsx": true,
    "useJSXTextNode": true
  },
  "extends": [
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:react/recommended", // Uses the recommended rules from @eslint-plugin-react
    "prettier/@typescript-eslint", // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
    "plugin:prettier/recommended" // Enables eslint-plugin-prettier and eslint-config-prettier. This will display prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
  ],
  "plugins": ["@typescript-eslint", "react-hooks", "react", "sort-imports-es6-autofix"],
  "rules": {
    "sort-imports-es6-autofix/sort-imports-es6": [
      1,
      {
        "ignoreCase": true,
        "ignoreMemberSort": false,
        "memberSyntaxSortOrder": ["none", "all", "multiple", "single"]
      }
    ],
    "@typescript-eslint/explicit-function-return-type": "off",
    "@typescript-eslint/no-use-before-define": "off",
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "argsIgnorePattern": "^_",
        "vars": "local",
        "args": "after-used",
        "ignoreRestSiblings": true
      }
    ],
    "react-hooks/rules-of-hooks": "error",
    "react-hooks/exhaustive-deps": "warn",
    "react/prop-types": "off",
    "react/display-name": "off",
    "import/no-unresolved": "off"
  },
  "settings": {
    "react": {
      "version": "detect" // Tells eslint-plugin-react to automatically detect the version of React to use
    }
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
useEffect(loadFamilyHistories, [, loadFamilyHistories])
```

```bash
eslint --ext .jsx,.js,.tsx,.ts .
```

**What did you expect to happen?**
Show an error|warn on vscode (other ide/editor) and/or bring error on the cli after run the _**eslint**_ command

**What actually happened? Please include the actual, raw output from ESLint.**
![image](https://user-images.githubusercontent.com/9604296/84829678-22ddd380-aff6-11ea-9dde-5c6372d01a71.png)

> eslint --ext .jsx,.js,.tsx,.ts .

Oops! Something went wrong! :(

ESLint: 7.2.0

TypeError: Cannot read property 'type' of null
Occurred while linting /home/leandrow/dev/wenlogic/ehr-app/app/js/modules/family-history/knowledge-area/KAFamilyHistoryContainer.tsx:21    at /home/leandrow/dev/wenlogic/ehr-app/node_modules/eslint-plugin-react-hooks/cjs/eslint-plugin-react-hooks.development.js:839:23
    at Array.some (<anonymous>)
    at visitCallExpression (/home/leandrow/dev/wenlogic/ehr-app/node_modules/eslint-plugin-react-hooks/cjs/eslint-plugin-react-hooks.development.js:838:86)
    at /home/leandrow/dev/wenlogic/ehr-app/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/leandrow/dev/wenlogic/ehr-app/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/home/leandrow/dev/wenlogic/ehr-app/node_modules/eslint/lib/linter/node-event-generator.js:254:26)
    at NodeEventGenerator.applySelectors (/home/leandrow/dev/wenlogic/ehr-app/node_modules/eslint/lib/linter/node-event-generator.js:283:22)
    at NodeEventGenerator.enterNode (/home/leandrow/dev/wenlogic/ehr-app/node_modules/eslint/lib/linter/node-event-generator.js:297:14)    at CodePathAnalyzer.enterNode (/home/leandrow/dev/wenlogic/ehr-app/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:635:23)

**Are you willing to submit a pull request to fix this bug?**
Could be

**Comment by mdjermanovic** [comment]
Hi @leandrowilliam, thanks for the issue!

By the provided call stack it seems to be a problem in third-party plugin `eslint-plugin-react-hooks`.

Can you please report the bug here:  https://github.com/facebook/react

**Comment by yeonjuan** [comment]
As @mdjermanovic mentioned, It's a bug in react.
I made a PR for fixing it. - https://github.com/facebook/react/pull/19145

**Comment by leandrowilliam** [comment]
thanks @mdjermanovic @yeonjuan 
closes by facebook/react#19145
